plugins/blynk-vhw/blynklib.py

```
@@ -1,211 +1,200 @@
+        except socket.timeout:
+            return b''
+        except socket.error as e:
+            if e.args[0] == EAGAIN:
+                return b''
+            else:
+                raise
+
+        if len(self.__rx_data) >= length:
+            data = self.__rx_data[:length]
+            self.__rx_data = self.__rx_data[length:]
+            return data
+        else:
+            return b''
+
+    def _ping(self):
+        self.__ping_time = int(time.time())
+        self.__ping_msq_id = self.get_msq_id()
+        self._send(HDR.pack(MsgType.PING, self.__ping_msq_id, 0)) 
+
+    def _loop(self):
+        while True:
+            self._connect()
+
+            while self.status == Status.AUTHENTICATED:
+                buffer = self._receive(HDR.size, 0.01)  
+                if buffer :
+                    msg_type, msg_id, data = HDR.unpack(buffer)
+                    if msg_type == MsgType.RSP:
+                        if msg_id == self.__ping_msq_id:
+                            self.__ping_msq_id = 0
+                    elif msg_type == MsgType.PING:
+                        self._send(HDR.pack(MsgType.RSP, msg_id, MsgStatus.OK))
+                    elif msg_type == MsgType.HW :
+                        params = (self._receive(data, 1).decode('ascii')).split("\0")
+                        if self.on_message :
+                            self.on_message(params)
+                else:
+                    if int(time.time()) - self.__ping_time >= 5 :
+                        if self.__ping_msq_id :
+                            print('Error: Blynk not receive ping response')
+                            self._close()
+                        else:
+                            self._ping()
+            time.sleep(1)
+            
+    def loop_forever(self):
+        while True:
+            try:
+                self._loop()
+            except KeyboardInterrupt:
+                sys.exit(0)
+            except:
+                pass 
```

I know that it works, but the point is that it’s very nasty – it just swallows exceptions without logging them and aborting the application when something bad happen. This is “exception handling” level PHP – just hide all exceptions and act like nothing happened. I really “love” troubleshooting such craps.